### PR TITLE
chore(MeshGatewayInstance): supplement creating validation for mode

### LINKED ISSUE #10276: AdmissionController should reject MeshGatewayInstance creation if the CP is global
### Description

In multiple zone mode(1 global-cp-k8s, 1 zone-cp-k8s), I thought that I should deploy the MeshGatewayInstance in the global-cp and sync it to the zones.

Improvements:
1. Supplement a validation for global-cp
2. Add docs clarification for multiple-zone builtin-gateway

### PR DESCRIPTION
**Pull request by Icarus9913**
AdmissionController should reject MeshGatewayInstance creation for Global CP.

close #10276 

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by Icarus9913** [comment]
WIP: Unit Test

**Review by michaelbeaumont — COMMENTED** [review]
I think better would be to not install the CRD at all

**Comment by Icarus9913** [comment]
> I think better would be to not install the CRD at all

That sounds like feasible and easier. But furthermore, the user may be curious that why is it that no CRD installed in the Global-CP and we need more obvious clue to remind the user of deploying the MeshGatewayInstance in the Zone-CP.
What do you think?

**Comment by lahabana** [comment]
@michaelbeaumont I think @Icarus9913 is right the user is gonna get it wrong. At least here they'll get a nice error message explaining them that they can't do that here.
Without the CRD they might just become suspicious of their install ("Maybe I didn't install CRDs correctly")

File: pkg/plugins/runtime/k8s/webhooks/gateway_instance_validator.go [hunk 1]
```diff
@@ -49,6 +55,11 @@ func (h *GatewayInstanceValidator) ValidateDelete(ctx context.Context, req admis
 }
 
 func (h *GatewayInstanceValidator) ValidateCreate(ctx context.Context, req admission.Request) admission.Response {
+	if h.cpMode == config_core.Global {
+		return admission.Denied(fmt.Sprintf("Operation not allowed. %s resources like %s can be created only from the %s control plane and not from a %s control plane.",
+			version.Product, "MeshGatewayInstance", strings.ToUpper(config_core.Zone), strings.ToUpper(h.cpMode)))
```
Comment (lahabana) [inline pkg/plugins/runtime/k8s/webhooks/gateway_instance_validator.go:60]:
```suggestion
			version.Product, "MeshGatewayInstance", config_core.Zone, h.cpMode))
```

You can emphasise with quotes instead

**Review by lahabana — APPROVED** [review]

**Comment by michaelbeaumont** [comment]
>  Without the CRD they might just become suspicious of their install ("Maybe I didn't install CRDs correctly")

I guess? But then they don't understand how Kuma/gateways work, which they could have figured out by once looking at the docs.
I think users assume they can actually use the CRDs that exist in their cluster, the existence of a CRD is an affordance. For example, this would cause problems with tooling that checks whether the CRD exists before making a decision to use it in some way.

To me it doesn't make sense, but we can merge it if you all think it improves things.

**Comment by jakubdyszkiewicz** [comment]
IMHO there's no harm in merging it.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**